community/kernel/src/main/java/org/neo4j/kernel/impl/transaction/log/PhysicalLogFileInformation.java

```diff
@@ -30,17 +31,17 @@
 
     private final PhysicalLogFiles logFiles;
     private final LogHeaderCache logHeaderCache;
-    private final TransactionIdStore transactionIdStore;
+    private final Supplier<Long> lastCommittedId;
     private final LogVersionToTimestamp logVersionToTimestamp;
 
     public PhysicalLogFileInformation( PhysicalLogFiles logFiles,
                                        LogHeaderCache logHeaderCache,
-                                       TransactionIdStore transactionIdStore,
+                                       Supplier<Long> lastCommittedId,
```

Comment:
What do you think about doing this generalization with inheritance instead:
```
               +--------------------------+
               |AbstractLogFileInformation|
               +--------+------------+----+
                        ^            ^
                        |            |
                        |            |
+-----------------------+-----+   +--+-------------------+
|TransactionLogFileInformation|   |RaftLogFileInformation|
+-----------------------------+   +----------------------+
```

Where `TransactionLogFileInformation` will still get `TransactionIdStore` in constructor and `RaftLogFileInformation` will do smth else.

Imo passing functions around like this makes code really complex.